src/core/server/http/http_server.ts

```diff
@@ -279,6 +280,23 @@ export class HttpServer {
     });
   }
 
+  private setupConditionalCompression() {
+    if (this.server === undefined) {
+      throw new Error('Server is not created yet');
+    }
+
+    this.server.ext('onRequest', (request, h) => {
```

Comment:
I wasn't able to use `registerOnPreAuth` here because it doesn't expose the `info`. I'm assuming this generally isn't something we want to support, so I'm using the HapiJS interfaces directly. If you all think we should modify the `KibanaRequest` to include this, just let me know.